### PR TITLE
Disable the spinner if CLI is not attached to terminal

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

### What this PR does / why we need it
Disable the spinner if CLI is not attached to terminal.
The previous forked branch `github.com/alonyb/spinner` has issue that causes spinner message being wiped out and redundant empty lines being printed out. Move the logic of checking terminal out to tanzu-framework

### Which issue(s) this PR fixes

Fixes # spinner message being wiped out and redundant empty lines being printed out

### Release note

```release-note
Bug fix: Fix the issue that spinner messages being wiped out and redundant empty lines being printed.
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -299,6 +297,8 @@ github.com/bombsimon/wsl/v2 v2.0.0/go.mod h1:mf25kr/SqFEPhhcxW1+7pxzGlW+hIl/hYTK
 github.com/bombsimon/wsl/v2 v2.2.0/go.mod h1:Azh8c3XGEJl9LyX0/sFC+CKMc7Ssgua0g+6abzXN4Pg=
 github.com/bombsimon/wsl/v3 v3.0.0/go.mod h1:st10JtZYLE4D5sC7b8xV4zTKZwAQjCH/Hy2Pm1FNZIc=
 github.com/bombsimon/wsl/v3 v3.1.0/go.mod h1:st10JtZYLE4D5sC7b8xV4zTKZwAQjCH/Hy2Pm1FNZIc=
+github.com/briandowns/spinner v1.16.0 h1:DFmp6hEaIx2QXXuqSJmtfSBSAjRmpGiKG6ip2Wm/yOs=
+github.com/briandowns/spinner v1.16.0/go.mod h1:QOuQk7x+EaDASo80FEXwlwiA+j/PPIcX3FScO+3/ZPQ=
```
Comment (vijaykatam) [inline go.sum:301]: Could we pick up latest release https://github.com/briandowns/spinner/releases/tag/v1.18.0?

**Review by vijaykatam — COMMENTED** [review]
Change LGTM. @maralavi could you help with verification?

File: go.sum (hunk 1 above)
Comment (blc1996) [reply go.sum:301]: The latest release would clean up the line after the spinner is stopped. https://github.com/briandowns/spinner/commit/12b31a684908a81e40889585a0fb69a5313a7eba#diff-7fa6384fd96c30fe4860e14a0f0d77214ebcc6af75b809031f21eb6a1cf0b1c7

Comment (vijaykatam) [reply go.sum:301]: Could we print a new line before spinner is stopped?

Comment (blc1996) [reply go.sum]: Actually, is it better to just eliminate each line after spinner stops? The spinner message will be shown during the process. The following are 2 types of output.

```
$ ./tanzu-core-darwin_amd64 package install my-ingress1 -p contour.tanzu.vmware.com -v 1.18.2+vmware.1-tkg.1-rc.4

Updated installed package 'my-ingress1'
```

```
$ ./tanzu-core-darwin_amd64 package install my-ingress1 -p contour.tanzu.vmware.com -v 1.18.2+vmware.1-tkg.1-rc.4
 Installing package 'contour.tanzu.vmware.com'
 Updating package 'my-ingress1'
 Getting package install for 'my-ingress1'

Updated installed package 'my-ingress1'
```

Comment (blc1996) [reply go.sum]: NVM, keeping the message looks better

**Comment by maralavi** [comment]
I just tested package plugin with Lucheng's fork and now I can see progress messages:

```
tanzu package install my-ingress -p contour.tanzu.vmware.com -v 1.18.2+vmware.1-tkg.1-rc.3
lucheng latest plugin changes
- Installing package 'contour.tanzu.vmware.com'I0114 15:31:14.913827   12625 request.go:665] Waited for 1.008797658s due to client-side throttling, not priority and fairness, request: GET:https://10.78.160.236:6443/apis/controlplane.cluster.x-k8s.io/v1alpha4?timeout=32s
 Installing package 'contour.tanzu.vmware.com'
 Getting package metadata for 'contour.tanzu.vmware.com'
 Creating service account 'my-ingress-default-sa'
 Creating cluster admin role 'my-ingress-default-cluster-role'
 Creating cluster role binding 'my-ingress-default-cluster-rolebinding'
 Creating package resource
 Waiting for 'PackageInstall' reconciliation for 'my-ingress'
 'PackageInstall' resource install status: Reconciling
 'PackageInstall' resource install status: ReconcileSucceeded
 'PackageInstall' resource successfully reconciled

 Added installed package 'my-ingress'
```

```
tanzu package installed update my-ingress -p contour.tanzu.vmware.com -v 1.18.2+vmware.1-tkg.1-rc.3
lucheng latest changes
 Updating installed package 'my-ingress'
\ Getting package install for 'my-ingress'I0114 15:41:33.645754   14054 request.go:665] Waited for 1.052559792s due to client-side throttling, not priority and fairness, request: GET:https://10.78.160.236:6443/apis/storage.k8s.io/v1?timeout=32s
 Getting package install for 'my-ingress'
 Getting package metadata for 'contour.tanzu.vmware.com'
Updated installed package 'my-ingress' in namespace 'default'
```

```
 tanzu package installed delete my-ingress                malavi@malavi-a01
Deleting installed package 'my-ingress' in namespace 'default'. Are you sure? [y/N]: y
lucheng latest plugin changes
 Uninstalling package 'my-ingress' from namespace 'default'
- Getting package install for 'my-ingress'I0114 15:33:58.645957   12686 request.go:665] Waited for 1.002164091s due to client-side throttling, not priority and fairness, request: GET:https://10.78.160.236:6443/apis/clusterctl.cluster.x-k8s.io/v1alpha3?timeout=32s
 Getting package install for 'my-ingress'
 Deleting package install 'my-ingress' from namespace 'default'
 'PackageInstall' resource deletion status: Deleting
 Deleting admin role 'my-ingress-default-cluster-role'
 Deleting role binding 'my-ingress-default-cluster-rolebinding'
 Deleting service account 'my-ingress-default-sa'
Uninstalled package 'my-ingress' from namespace 'default'
```

**Review by vijaykatam — APPROVED** [review]
LGTM

**Pull request merged**